File: backend/app/schemas/invitation.py
"""入职邀请出入参。"""
from pydantic import BaseModel, Field

from app.schemas.common import PhoneStr


class InvitationCreateIn(BaseModel):
    name: str | None = Field(default=None, max_length=64, description="被邀请人姓名（选填）")
    phone: PhoneStr
    email: str | None = Field(default=None, max_length=128)
    department_id: int = Field(description="预设部门（必填）")
    role_id: int = Field(description="预设角色（必填）")
    post: str | None = Field(default=None, max_length=64, description="预设岗位")
    expires_days: int = Field(default=3, ge=1, le=30, description="有效期（天）")
    remark: str | None = Field(default=None, max_length=255)


class InvitationAcceptIn(BaseModel):
    username: str = Field(min_length=3, max_length=64, pattern=r"^[a-zA-Z0-9_]+$")
    password: str = Field(min_length=6, max_length=64)

File: frontend/src/api/invitation.ts
import type { PageResult } from '@/types'
import { get, post, del } from './request'

export interface InvitationItem {
  id: number
  name: string
  phone: string | null
  email: string | null
  department_id: number | null
  department_name: string | null
  role_id: number | null
  role_name: string | null
  post: string | null
  token: string
  invite_link: string | null
  expires_at: string | null
  status: number
  status_label: string
  remark: string | null
  created_at: string
}

export interface InvitationLogItem {
  id: number
  action: string
  detail: string | null
  ip: string | null
  created_at: string
}

export const INVITATION_STATUS_COLOR: Record<number, string> = {
  0: 'default',
  1: 'processing',
  2: 'warning',
  3: 'success',
  4: 'default',
  5: 'error',
  6: 'error',
}

export const invitationApi = {
  list: (params: { status?: number; keyword?: string; page?: number; page_size?: number }) =>
    get<PageResult<InvitationItem>>('/invitations', { params }),
  create: (data: {
    name: string
    phone?: string
    email?: string
    department_id?: number
    role_id?: number
    post?: string
    expires_days?: number
    remark?: string
  }) => post<InvitationItem>('/invitations', data),
  resend: (id: number) => post<InvitationItem>(`/invitations/${id}/resend`),
  cancel: (id: number) => post<InvitationItem>(`/invitations/${id}/cancel`),
  remove: (id: number) => del(`/invitations/${id}`),
  logs: (id: number) => get<InvitationLogItem[]>(`/invitations/${id}/logs`),
  /** 公开：邀请信息与接受注册（无需登录） */
  publicInfo: (token: string) =>
    get<{
      name: string
      department_name: string | null
      role_name: string | null
      post: string | null
      status: number
      status_label: string
      expires_at: string | null
    }>(`/invitations/public/${token}`),
  publicAccept: (token: string, data: { username: string; password: string }) =>
    post<{ username: string; real_name: string }>(`/invitations/public/${token}/accept`, data),
}
